### DISCUSSION TITLE
Swiper 7

### DISCUSSION THREAD

**Discussion by nolimits4web** [nolimits4web/swiper#4228] [Ideas]
We need a smaller version of core swiper, as still current core version may contains a lot of functionality that doesn't required for everyone.

## Modules

List if settings/features (wip) to be potentially removed from core and moved to installable modules:
- [x] `freeMode` - a huge chunk in code in `onTouchEnd` handler related to `freeMode`
- [ ] `slidesPerColumn`
- [ ] `loop`
- [ ] `components/core/manipulation` - `appendSlide`/`prependSlide` and other methods which are totally not needed when used with Angular/React/Vue/Svelte components as manipulation should be done there by means of framework anyway

## Lite Version

In addition with need a "lite" version of Swiper that will have Core + most used modules like:
- navigation
- pagination
- lazy
- a11y
- autoplay

## Refactoring
- [ ] Rename Core `components` to `modules`

## Features
- [ ] add native `loading="lazy"` support to lazy loading module
- [x] auto something like `createEl: boolean` param to Navigation, Pagination, Scrollbar modules. And auto create elements if `el`'s are not passed

cc @vltansky

**Comment by nolimits4web** [comment]
For example, how `freeMode` can be extracted.

Basically we need a module that:
- adds `swiper.freeMode.onTouchMove` method with this logic: https://github.com/nolimits4web/swiper/blob/master/src/components/core/events/onTouchMove.js#L215-L227
- adds `swiper.freeMode.onTouchEnd` method where we move this chunk of logic: https://github.com/nolimits4web/swiper/blob/master/src/components/core/events/onTouchEnd.js#L77-L250
- adds this defaults to parameters: https://github.com/nolimits4web/swiper/blob/master/src/components/core/defaults.js#L26-L34

**Reply by vltansky** [reply]
But I see that now `freeMode` also prevents next chunk of code from execution:
https://github.com/nolimits4web/swiper/blob/master/src/components/core/events/onTouchEnd.js#L80

**Reply by nolimits4web** [reply]
Yes, so this part in onTouchEnd becomes something like:
```js
if (swiper.params.freeMode) {
  swiper.freeMode.onTouchEnd(...someparams);
  return;
}
```

**Reply by vltansky** [reply]
Ok, looks pretty easy.. I'll make a PR for `freeMode` and will see if I got it right :)

**Comment by vltansky** [comment]
Is removing features from Core means a Major release?
and if yes maybe this RFC should be converted to plans for major release (if there any big changes on your mind ofcourse)?

**Reply by nolimits4web** [reply]
For sure, it is going to be a major release. I think we need to create a `v7` branch for these changes

**Comment by ettoredn** [comment]
Why would lazy be needed when browsers have native lazy-loading capabilities?

**Reply by nolimits4web** [reply]
You probably will still want to show nice preloader :)

**Reply by ettoredn** [reply]
I believe this can still be done hooking into native DOM events (e.g. img.onload).

**Comment by ettoredn** [comment]
Please make `swiper.scss` support SASS modules, that is avoid it from outputting styles when `@use`'d. Support something like `@use 'swiper' with (output-styles: true)` to provide `@import` compatible behavior.

**Comment by leoalv** [comment]
Hi, I know this is just for ideas, but when is this version planned or calculated to be available?
It is that I was able to configure it with pagination and navigation outside the container and I love this, but the minified version weighs 142kb and left me unable to choose whether to use or change.

Thank you.

**Reply by ettoredn** [reply]
Are you using tree shaking?

**Reply by leoalv** [reply]
Hi @ettoredn what is tree shaking? I'm usign swiper-bundle.min

**Reply by ettoredn** [reply]
https://rollupjs.org/guide/en/#tree-shaking
